### PR TITLE
Improve IDEA integration

### PR DESCRIPTION
**Pull request by Glease**
* add a default project output path: Sometimes this is left empty somehow and IDEA complains if you try the native build. This gives a reasonable default.
* add option to override build type to gradle/native automatically: thanks raven for finding this idea-ext plugin. finally a headache solved.

Signed-off-by: Glease <4586901+Glease@users.noreply.github.com>

### CODE REVIEW

**Comment by Glease** [comment]
@eigenraven Hi, I need your help here. LWJGL libraries are marked as provided in IDEA (somehow), hence don't really show up when you use native IJ runs. However, if one does include these libraries in native IJ runs, you end up with compileOnly dependencies on classpath as well. What should we do here?
image

**Comment by eigenraven** [comment]
I have to find the documentation (or probably source code) of how intellij exactly converts gradle dependencies into its own. I keep all minecraft dependencies together in one "configuration" (patchedMinecraft) except lwjgl so that different run configurations can use different lwjgl versions, there's probably a mapping I could add somewhere to modify how IJ sees these.

File: build.gradle [hunk 1]
```diff
@@ -971,6 +976,38 @@ idea {
...
+                        options = ['delegatedBuild': 'false', 'testRunner': 'PLATFORM']
+                    else
+                        throw GradleScriptException('Accepted value for ideaOverrideBuildType is one of gradle or idea.')
+
+                    options.each { k, v ->
+                        gradleSettings.findFirstOrAppendChild('option', ['name': k]).attributes().value = v
+                    }
+                }
+            }
+            withIDEAFileXml("misc.xml") {
```
Comment (eigenraven) [inline build.gradle:1000]: this is something that needs further testing with the three IJ configurations: folder-based, file-based and implicit (no files) projects imported from gradle, iirc the xml functions were causing me trouble in one of these

Comment (Glease) [reply build.gradle:1000]: i would assume everyone is using imported from gradle, no? And if they don’t import from gradle this won’t run at all right?

Comment (eigenraven) [reply build.gradle:1000]: I meant this thing, though apparently file-based projects are now deprecated, I didn't realise it: <https://www.jetbrains.com/help/idea/creating-and-managing-projects.html#file-based>

**Pull request closed without merging**